compiler/src/dmd/init.d

```diff
@@ -38,6 +38,7 @@ extern (C++) class Initializer : ASTNode
 {
     Loc loc;
     InitKind kind;
+    bool semanticDone = false; /// initializerSemantic has been run on this
```

Comment:
Given this is a performance change, I would like to see some numbers about the tradeoff in adding additional members to a common ASTNode.

Alternately steal an unused bit from `kind` for this.

Reply:
For some internal dub projects at SARC I tested, it shaves off about 0.8s of debug build time (19.1s => 18.3s). I doubt increasing the class instance size from 21 to 22 is a problem, it probably gets rounded up to 24 anyways because of alignment, but I can measure memory usage later.

Reply:
I just noticed that `ArrayInitializer` already has a now redundant `sem` field, and `CInitializer` has an unused `sem` field. Since I removed them now, memory usage might even be reduced, since those field where actually at the end, while the new field sits in an alignment hole, since most derived classes start with a `Type` field with alignment 8.

Reply:
Oh I see, `InitKind : ubyte`